packages/presentation/lib/screens/login/login_bloc.dart

```diff
@@ -110,20 +127,72 @@ class _LoginBloc extends BlocImpl<BaseArguments, LoginData>
 
       _updateFieldControllers(user);
 
-      add(LoginData(
-        false,
+      add(state.copyWith(
+        authFailure: false,
       ));
 
       await _loginIfUserRegistered(user);
     } on AuthFailureException {
       add(
-        LoginData(
-          true,
+        state.copyWith(
+          authFailure: true,
         ),
       );
     }
   }
 
+  @override
+  bool validateForm() {
+    final user = UserEntity(
+      login: _loginController.text,
+      password: _passwordController.text,
+    );
+
+    final validationResult = _loginValidationUseCase(user);
+
+    add(
+      state.copyWith(
+        loginIsCorrect: validationResult.loginIsCorrect,
+        passwordIsCorrect: validationResult.passwordIsCorrect,
+      ),
+    );
+
+    return validationResult.loginIsCorrect &&
+        validationResult.passwordIsCorrect;
+  }
+
+  @override
+  String? loginValidator(
+    String wrongLoginMessage,
+    String requiredLoginMessage,
+  ) {
+    if (_loginController.text.isEmpty) {
```

Comment:
all validation logic mast be in use case as written in task: LoginValidationUseCase should contain all logic related to validation of both fields.

Reply:
Fixed.